COMPILING:
Latest commit on the drouin-simon:master branch did not compile on my Mac OSX El Capitan with Qt 5.6.1.

In file included from /Users/jbake/dev/boston/OpenIGTLinkIO/OpenIGTLinkIO/Examples/qIgtlClient/main.cpp:6:
In file included from /Users/jbake/dev/boston/OpenIGTLinkIO/OpenIGTLinkIO/GUI/qIGTLIOLogicController.h:7:
/Users/jbake/dev/boston/OpenIGTLinkIO/OpenIGTLinkIO/GUI/qIGTLIOVtkConnectionMacro.h:4:10: fatal error: 'vtkEventQtSlotConnect.h' file not found
#include "vtkEventQtSlotConnect.h"

Found out that vtkGUISupportQt needs to be an required component in the igtlioQtClient.

working:
find_package(VTK
  COMPONENTS
  vtkIOImage
  vtkImagingMath
  vtkGUISupportQt
  REQUIRED
  NO_MODULE)

not working:
find_package(VTK
  COMPONENTS
  vtkIOImage
  vtkImagingMath
  REQUIRED
  NO_MODULE)

CTEST:
When I added that line it compiled and I ran ctest which passed.

RUNNING:
This error which Christian reported is still there:
"B) When opening the example app igtlioQtClient, when adding a new Connector, the Devices view is not updated properly."

When I add more than one connector I get this output:
 ./bin/igtlioQtClient 
WARNING in: int qIGTLIOConnectorModel::rowCount(const QModelIndex& parent) const: Logic is a NULL object! Returning 0.
WARNING in: int qIGTLIOConnectorModel::rowCount(const QModelIndex& parent) const: Logic is a NULL object! Returning 0.
WARNING in: int qIGTLIODevicesModel::rowCount(const QModelIndex& parent) const: node is a NULL object! Returning 0.
WARNING in: int qIGTLIODevicesModel::rowCount(const QModelIndex& parent) const: node is a NULL object! Returning 0.

--- a/Examples/qIgtlClient/CMakeLists.txt
+++ b/Examples/qIgtlClient/CMakeLists.txt
@@ -13,6 +13,7 @@ find_package(VTK
   COMPONENTS
   vtkIOImage
   vtkImagingMath
+  vtkGUISupportQt
   REQUIRED
   NO_MODULE)
 include(${VTK_USE_FILE})

--- a/GUI/qIGTLIOVtkConnectionMacro.h
+++ b/GUI/qIGTLIOVtkConnectionMacro.h
@@ -9,8 +9,11 @@ void qvtkReconnect( vtkObject* old_vtk_obj, vtkObject* vtk_obj,             \
                     const char* qt_slot, float priority = 0.0,              \
                     Qt::ConnectionType connectionType = Qt::AutoConnection) \
 { \
-    MyQVTK.connectPtr->Disconnect( old_vtk_obj, vtk_event, qt_obj, qt_slot ); \
-    MyQVTK.connectPtr->Connect( vtk_obj, vtk_event, qt_obj, qt_slot, NULL, priority, connectionType ); \
+    Q_ASSERT( qt_obj ); \
+    if( old_vtk_obj ) \
+        MyQVTK.connectPtr->Disconnect( old_vtk_obj, vtk_event, qt_obj, qt_slot ); \
+    if( vtk_obj ) \
+        MyQVTK.connectPtr->Connect( vtk_obj, vtk_event, qt_obj, qt_slot, NULL, priority, connectionType ); \
 }
 
 

